storage/elasticsearch/elasticsearch_test.go

```diff
@@ -27,121 +27,122 @@ const testMapping = `
 }`
 
 func (c *ElasticSearchClient) indexEntry(id int) (bool, error) {
-	return c.Index("test_type", fmt.Sprintf("id%d", id), "{\"key\": \"val\"}")
+	return c.Index("test_type", fmt.Sprintf("id%d", id), `{"key": "val"}`)
 }
 
-func (c *ElasticSearchClient) cleanupIndices(name string) error {
-	if _, err := c.connection.DeleteIndex(fmt.Sprintf("%s_%s*", indexPrefix, name)); err != nil {
+func (c *ElasticSearchClient) cleanupIndices() error {
+	if _, err := c.connection.DeleteIndex(fmt.Sprintf("%s_%s*", indexPrefix, c.name)); err != nil {
 		return fmt.Errorf(fmt.Sprintf("Failed to clear test indices: %s", err.Error()))
 	}
 	return nil
 }
 
-func getClient(name string, limits ElasticLimits, mappings []map[string][]byte) (*ElasticSearchClient, error) {
-	client, err := NewElasticSearchClientFromConfig()
+func getClient(name string, indexCfg IndexConfig, mappings Mappings) (*ElasticSearchClient, error) {
+	connCfg := NewConnConfig("storage.elasticsearch")
+	client, err := NewElasticSearchClient(name, mappings, indexCfg, connCfg)
 	if err != nil {
 		return nil, err
 	}
-	if err := client.cleanupIndices(name); err != nil {
+	if err := client.cleanupIndices(); err != nil {
 		return nil, err
 	}
-	client.Start(name, mappings, limits)
+	client.Start()
 
 	return client, nil
 }
 
 // test rolling elasticsearch indices based on count limit
 func TestElasticsearchShouldRollByCount(t *testing.T) {
-	limits := ElasticLimits{}
-	limits.EntriesLimit = 5
+	indexCfg := IndexConfig{}
```

Comment:
please don't rename if it's not improving IndexConfig is exactly the same thing as ElasticLimits
let's focus on the patch behaviour, the review will be more readable